src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1852,6 +1855,16 @@ private AuthenticationResult getTokenWithRefreshToken(final IWindowComponent act
                 result.setRefreshToken(refreshItem.mRefreshToken);
             }
         } catch (IOException | AuthenticationException exc) {
+            Class causeClass = exc.getCause() != null ? exc.getCause().getClass() : exc.getClass();
```

Comment:
I agree, we should put the block of code separately in a method. If we're going to add more events logging, doing this way will produce more repeat code.